actors/builtin/paych/paych_test.go

```diff
@@ -45,63 +45,81 @@ func TestPaymentChannelActor_Constructor(t *testing.T) {
 		actor.constructAndVerify(t, rt, payerAddr, payeeAddr)
 	})
 
-	nonAccountCodeID := builtin.MultisigActorCodeID
+	t.Run("creates a payment channel actor after resolving non-ID addresses to ID addresses", func(t *testing.T) {
+		payerAddr := tutil.NewIDAddr(t, 101)
+		payerNonId := tutil.NewBLSAddr(t, 102)
+
+		payeeAddr := tutil.NewIDAddr(t, 103)
+		payeeNonId := tutil.NewBLSAddr(t, 104)
+
+		builder := mock.NewBuilder(ctx, paychAddr).
+			WithCaller(callerAddr, builtin.InitActorCodeID).
+			WithActorType(payerAddr, builtin.AccountActorCodeID).
+			WithActorType(payeeAddr, builtin.AccountActorCodeID)
+		rt := builder.Build(t)
+		rt.AddIDAddress(payerNonId, payerAddr)
+		rt.AddIDAddress(payeeNonId, payeeAddr)
+
+		actor.constructAndVerify(t, rt, payerNonId, payeeNonId)
+	})
+
 	testCases := []struct {
-		desc        string
-		fromCode    cid.Cid
-		fromAddr    addr.Address
-		toCode      cid.Cid
-		toAddr      addr.Address
-		expExitCode exitcode.ExitCode
+		desc               string
+		paymentChannelAddr addr.Address
+		callerCode         cid.Cid
+		newActorCode       cid.Cid
+		payerCode          cid.Cid
+		expExitCode        exitcode.ExitCode
 	}{
 		{"fails if target (to) is not account actor",
+			paychAddr,
+			builtin.InitActorCodeID,
+			builtin.MultisigActorCodeID,
 			builtin.AccountActorCodeID,
-			payerAddr,
-			nonAccountCodeID,
-			payeeAddr,
 			exitcode.ErrForbidden,
 		}, {"fails if sender (from) is not account actor",
-			nonAccountCodeID,
-			payerAddr,
+			paychAddr,
+			builtin.InitActorCodeID,
+			builtin.MultisigActorCodeID,
 			builtin.AccountActorCodeID,
-			payeeAddr,
 			exitcode.ErrForbidden,
-		}, {"fails if target cannot be resolved",
-			builtin.AccountActorCodeID,
-			tutil.NewSECP256K1Addr(t, "beach blanket babylon"),
-			builtin.AccountActorCodeID,
-			payeeAddr,
-			exitcode.ErrNotFound,
-		}, {"fails if sender cannot be resolved",
-			builtin.AccountActorCodeID,
-			payerAddr,
-			builtin.AccountActorCodeID,
-			tutil.NewSECP256K1Addr(t, "beach blanket babylon"),
-			exitcode.ErrNotFound,
 		},
 	}
 	for _, tc := range testCases {
 		t.Run(tc.desc, func(t *testing.T) {
 			builder := mock.NewBuilder(ctx, paychAddr).
-				WithCaller(callerAddr, builtin.InitActorCodeID).
-				WithActorType(paychAddr, builtin.PaymentChannelActorCodeID).
-				WithActorType(payerAddr, tc.toCode).
-				WithActorType(payeeAddr, tc.fromCode)
+				WithCaller(callerAddr, tc.callerCode).
+				WithActorType(tc.paymentChannelAddr, tc.newActorCode).
+				WithActorType(payerAddr, tc.payerCode)
 			rt := builder.Build(t)
 			rt.ExpectValidateCallerType(builtin.InitActorCodeID)
 			rt.ExpectAbort(tc.expExitCode, func() {
-				rt.Call(actor.Constructor, &ConstructorParams{To: tc.toAddr, From: tc.fromAddr})
+				rt.Call(actor.Constructor, &ConstructorParams{To: tc.paymentChannelAddr})
```

Comment:
What's going on here? The payment channel can't address itself. What happened to From?